src/test/app/AccountDelete_test.cpp

```diff
@@ -0,0 +1,699 @@
+            BEAST_EXPECT (! env.closed()->exists(keylet::offer (
+                carol.id(), carolOfferSeq)));
+            BEAST_EXPECT (! env.closed()->exists(keylet::signers (carol.id())));
+
+            // Verify that Carol's XRP, minus the fee, was transferred to becky.
+            BEAST_EXPECT (env.balance (becky) ==
+                carolOldBalance + beckyOldBalance - acctDelFee);
+        }
+    }
+
+    void testDirectories()
+    {
+        // The code that deletes consecutive directory entries uses a
+        // peculiarity of the implementation.  Make sure that peculiarity
+        // behaves as expected across owner directory pages.
+        using namespace test::jtx;
+
+        testcase("Directories");
+
+        Env env {*this};
+        Account const alice("alice");
+        Account const gw("gw");
+
+        env.fund(XRP(10000), alice, gw);
+        env.close();
+
+        // Alice creates enough offers to require two owner directories.
+        for (int i {0}; i < 45; ++i)
+        {
+            env (offer (alice, gw["USD"](1), XRP (1)));
+            env.close();
+        }
+        env.require (offers (alice, 45));
+
+        // Close enough ledgers to be able to delete alice's account.
+        std::uint32_t const ledgerCount {
+            openLedgerSeq (env) + 257 - env.seq (alice)};
+
+        for (std::uint32_t i = 0; i < ledgerCount; ++i)
+            env.close();
+
+        // Verify that both directory nodes exist.
+        Keylet const aliceRootKey {keylet::ownerDir (alice.id())};
+        Keylet const alicePageKey {keylet::page (aliceRootKey, 1)};
+        BEAST_EXPECT (env.closed()->exists (aliceRootKey));
+        BEAST_EXPECT (env.closed()->exists (alicePageKey));
+
+        // Delete alice's account.
+        auto const acctDelFee {drops (env.current()->fees().increment)};
+        auto const aliceBalance {env.balance (alice)};
+        env (acctdelete (alice, gw), fee (acctDelFee));
+        verifyDeliveredAmount (env, aliceBalance - acctDelFee);
+        env.close();
+
+        // Both of alice's directory nodes should be gone.
+        BEAST_EXPECT (! env.closed()->exists (aliceRootKey));
+        BEAST_EXPECT (! env.closed()->exists (alicePageKey));
+    }
+
+    void testOwnedTypes()
+    {
+        using namespace test::jtx;
+
+        testcase("Owned types");
+
+        // We want to test both...
+        //  o Old-style PayChannels without a recipient backlink as well as
+        //  o New-styled PayChannels with the backlink.
+        // So we start the test using old-style PayChannels.  Then we pass
+        // the amendment to get new-style PayChannels.
+        Env env {*this, supported_amendments() - fixPayChanRecipientOwnerDir};
+        Account const alice("alice");
+        Account const becky("becky");
+        Account const gw("gw");
+
+        env.fund(XRP(100000), alice, becky, gw);
+        env.close();
+
+        // Give alice and becky a whole passel of offers that we have to
```

Comment:
`passel` isn't a common word. I'm fine if you really want to keep it, but I'd probably lose it.

It is a cool word tho :-)